socorro/processor/cache_manager.py

```diff
@@ -554,12 +562,14 @@ def shutdown(self):
 
 
 def main():
-    from socorro.processor import cache_manager
-
-    app = cache_manager.DiskCacheManager()
+    app = DiskCacheManager()
     app.set_up()
     app.run_loop()
 
 
 if __name__ == "__main__":
-    main()
+    # NOTE(willkg): we need to do this so that the cache_manager logger isn't `__main__`
+    # which causes problems when logging
```

Comment:
The comment here is to remind me about how this affects logging which we have to explicitly set up loggers for. However, importing the code and running it runs the code in a context where `__name__` is set correctly. Further, if we move this code, then this line will break (correctly) and we'll update it. If we hard-coded the logger to be `socorro.processor.cache_manager`, then it wouldn't tell us it was wrong. Also, the `__name__` value which gets used in other places is still unhelpful.

We have this problem because we're not installing these scripts as a package or something like that. If we were installing these scripts, then the installation process creates a stub in some `bin/` directory like this:

```shell
#!/home/willkg/.pyenv/versions/3.10.9/bin/python
# -*- coding: utf-8 -*-
import re
import sys
from jsonschema.cli import main
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```

That imports the main function from the module, fixes `sys.argv[0]` (something we're not doing, so our sys.argv[0] is "python" which is fine), and then runs the main function.

I get that it's confusing, but I think this is the right thing to be doing for these services.

Reply:
That all makes sense. I still think having the same module in `sys.modules` twice creates its own pitfalls. One example is that all classes and functions in that module exist twice, so `isinstance()` might return `False` when it should return `True`. Maybe we can add our own minimal wrapper scripts and move all of the actual code to a module?

Reply:
A more realistic example would be this code
```
class MyException(Exception):
    pass


def main():
    raise MyException


if __name__ == "__main__":
    import a
    try:
        a.main()
    except MyException:
        print("MyException")
    except:
        print("other exception")
```
which prints "other exception".

I ran into this and similar issues before. Here's [an example I encountered recently](https://github.com/psf/requests/blob/9a40d1277807f0a4f26c9a37eea8ec90faa8aadc/src/requests/packages.py#L10-L14). It's not quite the same situation, though, since there could be substantial amounts of code using either version of the module while requiring interoperability.

Overall I agree with your reasoning, and it's not a problem here. It was just something I remembered as iffy, but the reasons don't apply here.